tableau_blueprints/refresh_datasource.py

```diff
@@ -0,0 +1,184 @@
+    from . import job_status
+
+EXIT_CODE_FINAL_STATUS_SUCCESS = 0
+EXIT_CODE_UNKNOWN_ERROR = 3
+EXIT_CODE_INVALID_CREDENTIALS = 200
+EXIT_CODE_INVALID_RESOURCE = 201
+EXIT_CODE_JOB_NOT_FOUND = 404
+EXIT_CODE_JOB_CACNELLED = 403
+EXIT_CODE_GENERIC_QUERY_JOB_ERROR = 400
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--username', dest='username', required=True)
+    parser.add_argument('--password', dest='password', required=True)
+    parser.add_argument('--site-id', dest='site_id', required=True)
+    parser.add_argument('--server-url', dest='server_url', required=True)
+    parser.add_argument('--datasource-name', dest='datasource_name', required=True)
+    parser.add_argument('--project-name', dest='project_name', required=True)
+    parser.add_argument('--check-status', dest='check_status', default='TRUE',
+                        choices={
+                            'TRUE',
+                            'FALSE'},
+                        required=False)
+    args = parser.parse_args()
+    return args
+
+
+def write_json_to_file(json_object, file_name):
+    with open(file_name, 'w') as f:
+        f.write(
+            json.dumps(
+                json_object,
+                ensure_ascii=False,
+                indent=4))
+    print(f'Response stored at {file_name}')
+
+
+def authenticate_tableau(username, password, site_id, server_url):
+    """TSC library to sign in and sign out of Tableau Server and Tableau Online.
+
+    :param username:The name of the user.
+    :param password:The password of the user.
+    :param site_id: The site_id for required datasources. ex: ffc7f88a-85a7-48d5-ac03-09ef0a677280
+    :param server_url: This corresponds to the contentUrl attribute in the Tableau REST API.
+    :return: connection object
+    """
+    try:
+        tableau_auth = TSC.TableauAuth(username, password, site_id=site_id)
+        server = TSC.Server(server_url, use_server_version=True)
+        server.version = '3.15'
+        connection = server.auth.sign_in(tableau_auth)
+    except Exception as e:
+        print(f'Failed to connect to Tableau.')
+        print(e)
+        sys.exit(EXIT_CODE_INVALID_CREDENTIALS)
+    return server, connection
+
+
+def get_datasource_id(server, connection, datasource_name, project_name):
+    """Returns the specified data source item
+
+    :param server:
+    :param connection: Tableau connection object
+    :param datasource_name: The name of the data source.
+    :param project_name: The name of the project associated with the data source.
+    :return: datasource : get the data source item to refresh
+    """
+    try:
+        datasource_id = None
+        with connection:
+            req_option = TSC.RequestOptions()
+            req_option.filter.add(TSC.Filter(TSC.RequestOptions.Field.Name,
+                                             TSC.RequestOptions.Operator.Equals,
+                                             datasource_name))
```

Comment:
Can you add another filter for project name to avoid the comparison in the loop?